src/main/java/cat/udl/eps/softarch/tfgfinder/controller/AuthenticationController.java

```diff
@@ -0,0 +1,43 @@
+package cat.udl.eps.softarch.tfgfinder.controller;
+
+import cat.udl.eps.softarch.tfgfinder.domain.*;
+import cat.udl.eps.softarch.tfgfinder.repository.UserRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/auth")
+public class AuthenticationController {
+
+    private final UserRepository userRepository;
+
+    public AuthenticationController(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @PostMapping("/register-student")
```

Comment:
This functionality is already implemented with the corresponding RestRepositories. It is even tested, for instance for student registration: https://github.com/UdL-EPS-SoftArch/tfgfinder-gei-api/blob/main/src/test/java/cat/udl/eps/softarch/tfgfinder/steps/RegisterStudentStepDefs.java#L78